### ISSUE TITLE
[SECURITY] URLs pointing to image files are automatically downloaded

### ISSUE THREAD

**Issue by Natureshadow** [siacs/Conversations#1122]
From #1121 , it was concluded that image links are, in fact, automatically downloaded by Conversations if they end in one of a set of known extensions. Allegedly, a fiel size limit is checked before the file is downloaded.

The latter does not help in any way, as the issue is of a different kind. Automatically sending an HTTP request to a random host upon receiving a message is a security flaw as it leaks private data to the HTTP server, e.g. if someone in a MUC sends an iamge link, at least a HEAD request is made to find out the file size, but that is enough to produce a connection to a foreign host which might be controlled by an attacker. The leaked data is more or less restricted to the client IP address, but especially on mobile networks, this is enough to retrieve rough location information as well.

The "feature" is enabled by default with the "Accept files smaller than …" option. File transfer and downloading image links should be split in two here, and downloading links should be disabled by default.

Opting out of automatic downloading of HTTP content is also a MUST in XEP-0071, which is not explicitly applicable here as it concerns XHTML-IM rather than links in plaintext messages, but I do not see any difference there: http://xmpp.org/extensions/xep-0071.html#security

**Comment by iNPUTmice** [comment]
This has been discussed in the original PR #428 
TLDR: images will only be accepted from contacts and not from strangers. the risks involved are basically the same as with jingle file transfer. (since jingle is P2P this will also leak your IP)

**Comment by Natureshadow** [comment]
> since jingle is P2P this will also leak your IP

In theory, yes. In practice, you will almost never see P2P transfers on mobile networks, as it simply doesn't work, and it will fall back to in-band byte streams.

**Comment by iNPUTmice** [comment]
> In theory, yes. In practice, you will almost never see P2P transfers on mobile networks, as it simply doesn't work, and it will fall back to in-band byte streams.

you can make it work if you are a malicious attacker
